### PR TITLE
resolve #13 - filterSuccess operator

### LINKED ISSUE #13: Add operators
```ts
todos$ = this.todosService.getTodos().pipe(filterSuccess()),
todos$ = this.todosService.getTodos().pipe(filterError())
todos$ = this.todosService.getTodos().pipe(select(result => result.data.foo))
```

### PR DESCRIPTION
**Pull request by rskendzic**

### CODE REVIEW

File: packages/playground/src/app/basic-page/basic-page.component.ts [hunk 1]
```diff
@@ -12,7 +13,7 @@ import { TodosService } from '../todos.service';
   template: `
     <h2 class="mb-3">Todos</h2>
 
-    <ng-container *subscribe="todos$ as todos">
+    <ng-container *ngIf="todos$ | async as todos">
```
Comment (NetanelBasal) [inline packages/playground/src/app/basic-page/basic-page.component.ts:16]: Why did you change it?

File: packages/operators/filter-success.operator.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Observable } from 'rxjs';
+import { NgQueryObserverResult } from '../ng-query/src/lib/types';
+
+export function filterSuccess() {
+  return function <T>(
```
Comment (NetanelBasal) [inline packages/operators/filter-success.operator.ts:5]: There is no need to create a new observable. You can use the filter operator.

File: packages/playground/src/app/basic-page/basic-page.component.ts [hunk 3]
```diff
@@ -53,7 +54,7 @@ import { TodosService } from '../todos.service';
 export class BasicPageComponent {
   private todosService = inject(TodosService);
   todo = new BehaviorSubject<number>(100);
-  todos$ = this.todosService.getTodos();
+  todos$ = this.todosService.getTodos().pipe(filterSuccess());
```
Comment (NetanelBasal) [inline packages/playground/src/app/basic-page/basic-page.component.ts:57]: There is no need to use it here. We want to show the default case.

File: packages/playground/src/app/basic-page/basic-page.component.ts (hunk 1 above)
Comment (rskendzic) [reply packages/playground/src/app/basic-page/basic-page.component.ts:16]: For some reason I had errors in IDE that properties weren't recognized. Will revert.

File: packages/ng-query/src/lib/operators/filter-success.operator.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { filter, Observable } from 'rxjs';
+import { NgQueryObserverResult } from '../types';
+
+export function filterSuccess() {
+  return function <T>(
+    source: Observable<NgQueryObserverResult<T>>
+  ): Observable<NgQueryObserverResult<T>> {
+    return source.pipe(filter((result) => result.status === 'success'));
```
Comment (NetanelBasal) [inline packages/ng-query/src/lib/operators/filter-success.operator.ts:8]: We'll want to infer the `data` has not nullable in this case—checkout filter with type guards.

Comment (rskendzic) [reply packages/ng-query/src/lib/operators/filter-success.operator.ts]: I have added type guard but not sure how would I do the same for the filterError operator

Comment (NetanelBasal) [reply packages/ng-query/src/lib/operators/filter-success.operator.ts]: Same, but `data` should be `null` and not `T`
